zerver/lib/narrow.py

```diff
@@ -755,11 +756,34 @@ def by_group_pm_with(
         return query.where(maybe_negate(cond))
 
     def by_search(self, query: Select, operand: str, maybe_negate: ConditionTransform) -> Select:
-        if settings.USING_PGROONGA:
+        if settings.USING_PG_SEARCH:
+            return self._by_search_pg_search(query, operand, maybe_negate)
+        elif settings.USING_PGROONGA:
             return self._by_search_pgroonga(query, operand, maybe_negate)
         else:
             return self._by_search_tsearch(query, operand, maybe_negate)
 
+    def _by_search_pg_search(
+        self, query: Select, operand: str, maybe_negate: ConditionTransform
+    ) -> Select:
+        # See search_operand_to_tantivy_query for details of what queries are supported.
+        text_search_query = search_operand_to_tantivy_query(operand)
+        safe_text_search_query = literal(text_search_query)
+
+        # TODO: finding the matches for higlighting is not implemented yet
+        query = query.add_columns(
+            literal_column("ARRAY[]::integer[]").label("content_matches"),
+            literal_column("ARRAY[]::integer[]").label("topic_matches"),
+        )
+
+        if text_search_query is None:
+            # The operand was not parsable or not supported. Return empty results
+            # from the search.
+            return query.where(false())
+
+        condition = literal_column("zerver_message.id").op("@@@")(safe_text_search_query)
```

Comment:
I believe this is the right way to do the queries in our context - this operator is documented at https://docs.paradedb.com/api-reference/optimization/overview The `fts_index.search(...)` operation shown in most of the docs is stated here as:
> While the [search function](https://docs.paradedb.com/api-reference/full-text/overview) is useful for simple full text search queries, it is not optimized for
>    Returning hundreds of thousands or millions of documents as quickly as possible
>   Usage in complex SQL queries with other “plain SQL” WHERE or JOIN clauses

In the narrows code we're doing a bunch of "plain SQL" filtering, so I think clearly this `@@@` operator is the way to go based on this doc.

A bit more detailed in their 0.10.0 release from last week where they started advertising this operator https://github.com/paradedb/paradedb/releases/tag/v0.10.0:
> using the @@@ operator in a WHERE clause, as opposed to SELECTing directly from idxname.search(...), can lead to drastically improved query execution times in situations where scores (or the effects of scoring on ordering) are not necessary..

Reply:
Cool, yeah I think this is the right design and you should just move a few sentences of that analysis into a comment above this line for posterity.

Reply:
Hi guys. In `v0.11.0`, releasing on Tuesday October 15, we are moving fully to the `@@@` operator. It is up to 100X faster on some queries. I would encourage you all to move to it, as `.search` will eventually be deprecated.